File: go/src/services/payment/paypal_service_adapter.go
package payment

import (
	"fmt"

	"github.com/csvitor-dev/PayTransaction/go/internal/entities"
	"github.com/fatih/color"
)

type PaypalServiceAdapter struct {
	service *paypalService
}

func NewPaypalServiceAdapter() *PaypalServiceAdapter {
	return &PaypalServiceAdapter{
		service: &paypalService{},
	}
}

func (adapter *PaypalServiceAdapter) ProcessPayment(payment entities.Payment) {
	color.HiYellow("-- Adapter [start] --")
	fmt.Println()

	paymentId := fmt.Sprint(payment.Id)
	amount := float64(payment.Amount)

	adapter.service.pay(paymentId, amount)

	color.HiGreen("Payment successfully processed.")
	fmt.Println(payment.Description)

	fmt.Println()
	color.HiYellow("-- Adapter [end] --")
}

File: ts/src/services/paypal-service-adapter.ts
import { PaypalService } from "./paypal-service";
import { PaymentService } from "../domain/contracts/payment-service";
import { Payment } from "../domain/models/payment";

export class PaypalServiceAdapter implements PaymentService {
  public constructor(private readonly service: PaypalService) {}

  processPayment(payment: Payment): void {
    console.log("-- Adapter [start] --");

    const { amount } = payment;
    const paymentId = payment.id.toString();

    this.service.pay(paymentId, amount);

    console.log("Payment successfully processed.");
    console.log(payment.description);

    console.log("-- Adapter [end] --");
  }
}
